pkg/stanza/fileconsumer/util_test.go

```diff
@@ -76,14 +76,24 @@ type emitParams struct {
 	token []byte
 }
 
-func buildTestManager(t *testing.T, cfg *Config) (*Manager, chan *emitParams) {
+func buildTestManager(t *testing.T, cfg *Config, initializeChannel bool) (*Manager, chan *emitParams) {
```

Comment:
Rather than require every caller to pass an additional arg, can we refactor `buildTestManager` and `buildTestManagerWithEmit` into `buildTestManager(t *testing.T, cfg *Config, opts ...testManagerOption)`? Something like:

```go
type testManagerConfig {
    emitChan chan *emitParams
    readerChan bool
} 

type testManagerOption func(*testManagerConfig)

func withEmitChan(emitChan chan *emitParams) testManagerOption {
    return func(m *testManagerConfig) {
        m.emitChan = emitChan
    }
}

func withReaderChan() testManagerOption {
    return func(m *testManagerConfig) {
        m.readerChan = true
    }
}

func buildTestManagerWithOptions(t *testing.T, cfg *Config, opts ...testManagerOption) (*Manager, chan *emitParams) {
    tmc := &testManagerConfig{ emitChan: make(chan *emitParams, 100) }
    for _, opt := range opts {
        opt(tmc)
    }
    input, err := cfg.Build(testutil.Logger(t), testEmitFunc(tmc.emitChan))
    require.NoError(t, err)
    if tmc.readerChan {
		input.readerChan = make(chan ReaderWrapper, cfg.MaxConcurrentFiles/2)
		ctx, cancel := context.WithCancel(context.Background())
		input.cancel = cancel
		input.ctx = ctx
		for i := 0; i < cfg.MaxConcurrentFiles/2; i++ {
			input.workerWg.Add(1)
			go input.worker(input.ctx)
		}
	}
    return m, nil
}
```

Reply:
Done, please review.